src/components/Md5Form.test.js

```diff
@@ -0,0 +1,70 @@
+/* eslint-disable testing-library/no-node-access */
+/* eslint-disable testing-library/no-container */
+import { screen, render, waitFor } from '@testing-library/react';
+import Md5Form from './Md5Form';
+import { getMd5 } from '../providers/md5Provider';
+import userEvent from '@testing-library/user-event';
+
+jest.spyOn(window, 'fetch')
+
+describe('Md5Form component', () => {
+    it('find typed text within element of class .data-text', () => {
+        render(<Md5Form getMd5={getMd5}/>)
+        const input = screen.getByRole('textbox');
+        const testText = 'test';
+        expect(input).toBeInTheDocument();
+        userEvent.type(input, testText);
+        const classElement = screen.getByText(testText, {className: 'data-text'})
+        expect(classElement).toBeInTheDocument()
+    });
+    it('should render data in md5Element on submit form', async () => {
+        
+        const {container} = render(<Md5Form getMd5={getMd5}/>)
+        const button = screen.getByText('send');
+        const input = screen.getByRole('textbox');
+
+        const md5Element = container.querySelector('.data-md5');
+
+        expect(button).toBeInTheDocument();
+
+        expect(md5Element).toBeInTheDocument();
+        
+        window.fetch.mockResolvedValueOnce({
+            ok: true,
+            json: async () => {
+                return {Digest: 'test1'}
```

Comment:
Raczej starałbym się bardziej realne dane tutaj wstawić :)
Dobrze jest też zapisać do zmiennej, aby móc to potem wykorzystać przy sprawdzaniu tj. linii 41 - wtedy nie zrobimy literówki :P